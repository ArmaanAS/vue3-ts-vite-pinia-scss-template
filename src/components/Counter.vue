<script lang="ts" setup>
import { toRefs } from "vue";

const props = defineProps<{ count: number }>();
const { count } = toRefs(props);
const emit = defineEmits(["update:count"]);

function increment() {
  emit("update:count", count.value + 1);
}
</script>

<template>
  <button type="button" @click="increment">count is {{ count }}</button>
</template>

<style lang="scss"></style>
