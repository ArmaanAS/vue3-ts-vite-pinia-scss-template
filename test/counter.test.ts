import { test, expect } from "vitest";
import { mount } from "@vue/test-utils";
import Counter from "../src/components/Counter.vue";

test("mount component", async () => {
  expect(Counter).toBeTruthy();

  const wrapper = mount(Counter, {
    props: {
      "count": 4,
    },
  });

  expect(wrapper.text()).toContain("4");
  expect(wrapper.html()).toMatchSnapshot();

  await wrapper.get("button").trigger("click");
});