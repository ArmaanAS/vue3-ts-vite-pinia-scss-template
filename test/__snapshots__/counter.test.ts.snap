// Vitest Snapshot v1

exports[`mount component 1`] = `"<button type=\\"button\\">count is 4</button>"`;
